zellij-server/src/lib.rs

```diff
@@ -278,7 +278,8 @@ pub fn start_server(mut os_input: Box<dyn ServerOsApi>, socket_path: PathBuf) {
                 // set the sticky bit to avoid the socket file being potentially cleaned up
                 // https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html states that for XDG_RUNTIME_DIR:
                 // "To ensure that your files are not removed, they should have their access time timestamp modified at least once every 6 hours of monotonic time or the 'sticky' bit should be set on the file. "
-                set_permissions(&socket_path, 0o1700).unwrap();
+                // It is not guaranteed that all platforms allow setting the sticky bit on sockets!
+                drop(set_permissions(&socket_path, 0o1700));
```

Comment:
I'd prefer to fix this by not setting the sticky bit explicitly on freebsd via conditional compilation, sound reasonable?

Reply:
In that case e.g. running the Linux binary on FreeBSD and other places (WSL1?) via an ABI compatibility layer might fail in the same way… This call is really not critical at all, I think it should be at most a warning, I don't think it could ever be crash-worthy!